combinetf2/physicsmodels/basemodel.py

```diff
@@ -0,0 +1,82 @@
+import tensorflow as tf
+
+
+class Basemodel:
+    """
+    A class to output histograms without any transformation, can be used as base class to inherit custom physics models from.
+    """
+
+    name = "basemodel"
+    need_observables = True  # if observables should be provided to the compute function
+    need_params = (
```

Comment:
Is there really any advantage to doing this?  The code is arguably simpler if all models take both the observables and the parameters and then just don't use one or the other if not needed.

Reply:
This is to avoid computing the observables if they are not needed.
I checked, it's running both the tracing and the computation through the flat_fun() also if observables are not used further.
I will simplify that params are always provided since they don't have to be computed.